e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,249 @@
+        public void T1_Browse_Address_Space() {
+            var endpoints = ListEndpointsAsync().GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.First();
+
+            Assert.NotNull(endpoint.Id);
+            Assert.Equal("ActivatedAndConnected", endpoint.ActivationState);
+            Assert.Equal("Ready", endpoint.EndpointState);
+
+            var nodes = GetNodesAsync(endpoint.Id).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+        }
+
+        [Fact, PriorityOrder(4)]
+        public void T1_1_BrowseSpecificNode() {
+            var endpoints = ListEndpointsAsync().GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.First();
+
+            Assert.NotNull(endpoint.Id);
+            Assert.Equal("ActivatedAndConnected", endpoint.ActivationState);
+            Assert.Equal("Ready", endpoint.EndpointState);
+
+            var nodes = GetNodesAsync(endpoint.Id).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            var nodeId = nodes.First().NodeId;
+
+            nodes = GetNodesAsync(endpoint.Id, nodeId).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+        }
+
+        [Fact, PriorityOrder(5)]
+        public void T1_2_BrowseAllNodesOfTypeObject() {
+            var endpoints = ListEndpointsAsync().GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.First();
+
+            Assert.NotNull(endpoint.Id);
+            Assert.Equal("ActivatedAndConnected", endpoint.ActivationState);
+            Assert.Equal("Ready", endpoint.EndpointState);
+
+            var nodes = GetAllNodesAsync(endpoint.Id, "Object").GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            Assert.True(nodes.Count > 150);
+        }
+
+        [Fact, PriorityOrder(6)]
+        public void T1_3_BrowseAllNodesOfTypeVariable() {
+            var endpoints = ListEndpointsAsync().GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.First();
+
+            Assert.NotNull(endpoint.Id);
+            Assert.Equal("ActivatedAndConnected", endpoint.ActivationState);
+            Assert.Equal("Ready", endpoint.EndpointState);
+
+            var nodes = GetAllNodesAsync(endpoint.Id, "Variable").GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            Assert.True(nodes.Count > 150);
+        }
+
+        /// <summary>
+        /// Equivalent to GetSetOfUniqueNodesAsync.
+        /// </summary>
+        private async Task<List<(string NodeId, string NodeClass, bool Children)>> GetNodesAsync(string endpointId, string nodeId = null, string continuationToken = null) {
```

Comment:
You could also move this methods to TestHelper if they are also usable/useful by other E2E tests

Reply:
Moved to TestHelper